.github/workflows/training-e2e.yaml

```diff
@@ -27,9 +27,48 @@ concurrency:
   cancel-in-progress: false
 
 jobs:
+  block-pr:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if PR is from forked repo
+        id: check_forked
+        run: |
+          if [[ ${{ github.event.pull_request.head.repo.full_name }} != ${{ github.event.pull_request.base.repo.full_name }} ]]; then
+            echo "This pull request was pushed from a forked repository, which is not allowed. Please push changes from a branch in the upstream repository"
+            exit 1
+          else
+            echo "This PR is from upstream repo."
+          fi
+
+  check-membership:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check if user is a member of the organization
+        id: check_membership
+        run: |
+          ORG="your-organization"
+          USER=${{ github.actor }}
+          RESPONSE=$(curl -s -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/orgs/containers/members/$USER")
```

Comment:
Would that work if this pipeline triggered from a forked PR? Specifically asking about the secrets.GITHUB_TOKEN.

Reply:
Maybe there is a GitHub Actions predefined variable that has this value already populated

Reply:
so GITHUB_TOKEN should be a predefined variable defined in Github. So usually we do not set these.